File: backend/responses/libraryListsResponse.py
from typing import List
from pydantic import BaseModel

from backend.responses.rockItAlbumWithoutSongsResponse import RockItAlbumWithoutSongsResponse
from backend.responses.rockItPlaylistResponse import RockItPlaylistResponse


class LibraryListsResponse(BaseModel):
    albums: List[RockItAlbumWithoutSongsResponse]
    playlists: List[RockItPlaylistResponse]

File: src/responses/libraryListsResponse.ts
import * as z from "zod";
import { RockItAlbumWithoutSongsResponse } from "./rockItAlbumWithoutSongsResponse";
import { RockItPlaylistResponse } from "./rockItPlaylistResponse";

export const LibraryListsResponse = z.object({
    albums: z.array(RockItAlbumWithoutSongsResponse),
    playlists: z.array(RockItPlaylistResponse),
});

export type LibraryListsResponse = z.infer<typeof LibraryListsResponse>;
